cowprotocol/services
# [Driver] Allow stale liquidity for quote requests

# Description
When giving production traffic to the colocated setup, we noticed a lot of price estimators (even fast ones like baseline) getting rate limited frequently. This is due to the driver requesting liquidity for the most recent block and blocking on fetching it if it's not available. For the legacy setup, there was a code path specifically for quotes, allowing to use a "recent" instead of the latest block for fetching liquidity.

This PR recreates this path for the co-located setup

# Changes


- Adds a flag to the liquidity fetcher indicating whether stale liquidity is allowed
- If this flag is set, quote for _recent_ instead of _latest_ block (which should already be cached)

## How to test
Run this benchmark script against before and after:

```sh
SECONDS=0 
while (( SECONDS < 60 )); do
  time curl -H 'content-type: application/json' --data '{"from": "0xc3792470cee7e0d42c2be8e9552bd651766c5178","buyToken": "0xa0b86991c6218b36c1d19d4a2e9eb0ce3606eb48","sellToken": "0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2","kind": "sell","sellAmountAfterFee": "100000000000000000", "quality":"optimal"}' http://localhost:8080/api/v1/quote
done
```

observe http://localhost:11088/metrics in both cases and look for liquidity cache hits:

Before:
```
driver_recent_block_cache_hits{cache_type="uniswapv2"} 34300
driver_recent_block_cache_misses{cache_type="uniswapv2"} 840
```

After
```
driver_recent_block_cache_hits{cache_type="uniswapv2"} 39200
driver_recent_block_cache_misses{cache_type="uniswapv2"} 140
```

Note that we now only have 1 cache miss (140 pools) on cold start vs 1 cache miss on each new block and higher overall throughput

## Related issues
#1672
